From 8f131de086c1527700ef8b05fb3c2e154824dec8 Mon Sep 17 00:00:00 2001
From: Carles <cpbadosa@gmail.com>
Date: Sun, 19 Mar 2017 15:22:57 +0100
Subject: [PATCH] Make sure string2btn matches btn2string's output

btn2string returns "DPAD_UP" when passed XBOX_DPAD_UP and gives similar
responses for the other directions (DPAD_LEFT, DPAD_RIGHT and DPAD_DOWN).
However, string2btn didn't accept those as input, rendering the output of
--help-button confusing.
---
 src/xboxmsg.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/xboxmsg.cpp b/src/xboxmsg.cpp
index 7abc64f3..48ff868d 100644
--- a/src/xboxmsg.cpp
+++ b/src/xboxmsg.cpp
@@ -1283,13 +1283,13 @@ XboxButton string2btn(const std::string& str_)
   else if (str == "tr" || str == "r3")
     return XBOX_BTN_THUMB_R;
 
-  else if (str == "du" || str == "up")
+  else if (str == "du" || str == "up" || str == "dpad_up")
     return XBOX_DPAD_UP;
-  else if (str == "dd" || str == "down")
+  else if (str == "dd" || str == "down" || str == "dpad_down")
     return XBOX_DPAD_DOWN;
-  else if (str == "dl" || str == "left")
+  else if (str == "dl" || str == "left" || str == "dpad_left")
     return XBOX_DPAD_LEFT;
-  else if (str == "dr" || str == "right")
+  else if (str == "dr" || str == "right" || str == "dpad_right")
     return XBOX_DPAD_RIGHT;
 
   else
